sklearn/linear_model/_ridge.py

```diff
@@ -2038,6 +2035,33 @@ def fit(self, X, y, sample_weight=None):
         the validation score.
         """
         cv = self.cv
+
+        check_scalar_alpha = partial(
+            check_scalar,
+            target_type=numbers.Real,
+            min_val=0.0,
+            include_boundaries="neither",
+        )
+
+        if isinstance(self.alphas, collections.abc.Collection):
```

Comment:
Let's revert this part to be in accordance with the reviewed done there: https://github.com/scikit-learn/scikit-learn/pull/21341/files#diff-691236f4342ae8b778b880831001f0d7838c917a0eb3e1d7465d9e5db1a10d95R564

Sorry about that.

Reply:
Thanks! Done!